thirdparty/DiskANN/src/index.cpp

```diff
@@ -226,7 +226,7 @@ namespace diskann {
   }
 
   template<typename T, typename TagT>
-  Index<T, TagT>::Index(Metric m, const size_t dim, const size_t max_points,
+  Index<T, TagT>::Index(Metric m, bool ip_prepared, const size_t dim, const size_t max_points,
```

Comment:
Can I ask why we don'y simply use `Metrics m` as the indicator?

Reply:
![image](https://github.com/milvus-io/knowhere/assets/11503321/7ff77cdf-d9c3-48cb-98e3-b29ac14a8d5e)
metric is L2 all the time even for IP.